### PR TITLE
Configure autoreleases

### PR DESCRIPTION
**Pull request by rtpg**
So according to https://github.com/jazzband-roadies/help/issues/126 it _sounded like_ we had configured autoreleases for this project, but I don't think we actually did.

@jezdez I might need help with the password field on this travis config, please feel free to directly edit the YAML in the right way. Also @jdufresne and I talked a bit through e-mail, I think you might need to set the lead for this project in the jazzband infra. I think he + I as two leads would give good coverage.

Apologies for the increased chatter, trying to close out getting a release on this lib and after all this everything should go smoothly :)

### CODE REVIEW

**Comment by jezdez** [comment]
@auvipy Heeeey, so this PR does not look good since it's obviously missing the password to have the release process work. I don't think this should have been gotten your review approval.

**Comment by jezdez** [comment]
@rtpg Thanks for the ping, that's okay, there are a few of project setups that are unfinished, gonna work on it now.

**Comment by jezdez** [comment]
@rtpg @jdufresne I've added you both as project leads now and enable the project releases.

While we're there, I think it would make sense to also starting using [tox-travis](https://github.com/tox-dev/tox-travis) like in other Jazzband projects since it simplifies the Travis config a lot and maybe use Travis [Build Stages](https://docs.travis-ci.com/user/build-stages/) to only release when all other test environments have passed. That would remove the hard-coding of the TOX_ENV to push it.

**Comment by auvipy** [comment]
I am not following the jazzband projects closed lately. but @rtpg reached me via email if im PL. i give approval but didn't merge as there wasn't a password.

**Comment by rtpg** [comment]
Alrighty, will see if I can get `tox-travis` set up nicely here

**Comment by rtpg** [comment]
OK, configured `tox-travis` + used jobs instead of the matrices. There's a bit of weirdness in the jobs where I feel like it's not respecting `envlist` in `tox.ini`.... but not really sure how to tell travis "run this only for these kinds (plural) of envs"

If anyone knows how to make that jobs config better I'd appreciate the help. If not, I think it's fine for us to move forward with this as-is.

File: .travis.yml [hunk 1]
```diff
@@ -1,52 +1,48 @@
...
+      env:
+      python: 3.8
+      script: skip
+      deploy:
+        provider: pypi
+        user: jazzband
+        distribution: sdist bdist_wheel
+        server: https://jazzband.co/projects/django-taggit/upload
+        password:
+            secure: ky+UZnCbKb213Ohqnv2uEfef3vXv1JoAaG5vCjR6Xspf+Ahz4F/4oKe5Noo/xxnRs1x8bjDfYFIbQVQYKyM95dJ7VVQWtrwjId+NMCE9K36l1qC199eCUxkYd/jPg6iM8no4gXa3AOyam+j2yMhAgqfM1YKMNFW7j1kR+T7cz4A=
```
Comment (jezdez) [inline .travis.yml:36]: Mind using 2 character indentations here please?

File: .travis.yml [hunk 2]
```diff
@@ -1,52 +1,48 @@
...
+        on:
+            tags: true
```
Comment (jezdez) [inline .travis.yml:38]: Same here.

File: .travis.yml [hunk 3]
```diff
@@ -1,52 +1,48 @@
...
+  - DJANGO=master
+
+jobs:
   fast_finish: true
+  allow_failures:
+    - env: DJANGO=master
   include:
-    - env: TOXENV=black
+    - python: 3.8
+      env: TOXENV=black
```
Comment (jezdez) [inline .travis.yml:21]: I think you'll have to specify the Python version for each of the extra environments, not only for black, to make sure we run them on 3.8, e.g. flake8 and isort are run on 3.5 at the moment.

**Comment by rtpg** [comment]
OK build didn't complete just yet but I think it's good

I couldn't figure out the format for the `allow_failures` thing in this situation. I think that testing on `master` is still valuable, and given Django's PR merge policies, failures on `master` will likely indicate real issues that we can tackle, so having the failures feels liike useful info.

**Comment by jezdez** [comment]
> OK build didn't complete just yet but I think it's good
> 
> I couldn't figure out the format for the `allow_failures` thing in this situation. I think that testing on `master` is still valuable, and given Django's PR merge policies, failures on `master` will likely indicate real issues that we can tackle, so having the failures feels liike useful info.

Yeah, I understand, let's roll with that and see if it works for us..

**Comment by rtpg** [comment]
Alright got coverage reports working nicely again too, going to merge this in. Thanks for the help, everyone

**Pull request merged**